llvm/test/CodeGen/PowerPC/aix-tls-gd-double.ll

```diff
@@ -612,14 +612,14 @@ entry:
 
 ; External symbol reference checks for .__tls_get_addr/.__tls_get_mod
 
-; SMALL32: .extern .__tls_get_addr[PR]
-; SMALL32: .extern .__tls_get_mod[PR]
-; SMALL64: .extern .__tls_get_addr[PR]
-; SMALL64: .extern .__tls_get_mod[PR]
-; LARGE32: .extern .__tls_get_addr[PR]
-; LARGE32: .extern .__tls_get_mod[PR]
-; LARGE64: .extern .__tls_get_addr[PR]
-; LARGE64: .extern .__tls_get_mod[PR]
+; SMALL32: .extern .{{__tls_get_addr|__tls_get_mod}}[PR]
```

Comment:
Could you please explain why the order changes with/without your change about supporting local dynamic TLS mode? Thanks very much.

Reply:
Before the LD patch there is only .__tls_get_addr[PR]
After the LD patch, there is .__tls_get_addr[PR] and .__tls_get_mod[PR]

The LD patch turned some variables from GD to LD, so created new extern

Reply:
OK. One more question is do we know why TLS external function calls are impacted by the iteration order, while for below cases:
```
int foo();
int bar();

int test() {
  return foo() + bar();
}
```

The external `foo()` and `bar()` are the same with both iteration order.

Could you please help to figure that out? And yes, like @efriedma-quic , we may need to fix this in the compiler, not the test case.

Thanks a lot.